lxd/main_forkconsole.go

```diff
@@ -74,6 +75,14 @@ func (c *cmdForkconsole) Run(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("Error opening config file: %q", err)
 	}
 
+	rawConfigPath := fmt.Sprintf("%s.raw", configPath)
```

Comment:
This repetition everywhere that d.LoadConfigFile() is used feels too easy to miss when someone uses `LoadConfigFile` in the future and misses loading the raw file too. Can we have a utility function that contains this logic in one place and replaces all use of `d.LoadConfigFile()` in the LXD code base?

Reply:
Also as a "nice to have", we could make go-lxcs `LoadConfigFile` accept a variadic argument indicating you can pass multiple paths and avoid the duplicated function call.

Reply:
The latter option sounds good. The former can be done but we'd want to remain in-package as we're trying to keep those forkXYZ codepaths as small as possible and need minimal involvement of goroutines and the like.

Reply:
@tomponline I've actually updated the branch now to only load raw.lxc during startup (forkstart/forkmigrate), the other cases should not need it, so it's actually more consistent to just not do it then (otherwise we'd need the C-only forkexec/forkmount to also do it).

Reply:
In that case can it just stay being generated just startCommon rather than being added as two separate files, and only loaded in the scenarios where startCommon would be called anyway?

Reply:
So based on your comment https://github.com/lxc/lxd/pull/9094#commitcomment-54520855 the fix you added in `Update()` handles the validation issue. Then we dont need separate files as we can just only add the raw.lxc at the end in startCommon (for start and migrate scenarios you mention). As for calling it twice, I think other things would also fail in that case, like duplicate device setup conflicts, to not need to worry about that.

Reply:
@tomponline updated